python_modules/dagster/dagster/core/execution/plan/execute_step.py

```diff
@@ -523,7 +525,11 @@ def _gen_fn():
         ),
         handle_output_gen,
     ):
-        if isinstance(elt, AssetMaterialization):
+        for event in output_context.consume_events():
+            yield event
+        if isinstance(elt, DagsterEvent):
+            yield elt
```

Comment:
what case is this exactly? Is it under test?

Reply:
We create the `DagsterEvent` immediately upon it being logged, and it gets yielded at a layer under this. So we need to make sure we can bubble up DagsterEvents properly. It's under test just by using the logging methods

Reply:
aren't the events from using the logging methods handled by the yield above using `consume_events`? I don't see when `elt` will be a `DagsterEvent` in the test case